src/main/scala/org/broadinstitute/dsde/firecloud/webservice/UserApiService.scala

```diff
@@ -83,43 +83,43 @@ trait UserApiService extends HttpService with PerRequestCreator with FireCloudRe
+                  case NotFound => respondWithErrorReport(NotFound, "FireCloud user registration not found.", requestContext)
                   // rawls error? boo. All we can do is respond with an error.
-                  case InternalServerError => respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
+                  case InternalServerError => respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again. (Error 10)", requestContext)
                   // rawls found the user; we'll try to parse the response and inspect it
                   case OK =>
                     val respJson = response.entity.as[RegistrationInfo]
                     respJson match {
                       case Right(regInfo) =>
                         if (regInfo.enabled.google && regInfo.enabled.ldap && regInfo.enabled.allUsersGroup) {
-                          // rawls says the user is fully registered and activated!
+                          // Sam says the user is fully registered and activated!
                           requestContext.complete(OK, regInfo)
                         } else {
-                          // rawls knows about the user, but the user isn't activated
-                          respondWithErrorReport(Forbidden, "FireCloud user not activated", requestContext)
+                          // Sam knows about the user, but the user isn't activated
+                          respondWithErrorReport(Forbidden, "FireCloud user not activated.", requestContext)
                         }
-                      // we couldn't parse the rawls response. Respond with an error.
+                      // we couldn't parse the Sam response. Respond with an error.
                       case Left(error) =>
-                        respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
+                        respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again. (Error 20)", requestContext)
                     }
                   case x =>
-                    // if we get any other error from rawls, pass that error on
+                    // if we get any other error from Sam, pass that error on
                     respondWithErrorReport(x.intValue, "Unexpected response validating registration: " + x.toString, requestContext)
                 }
-              // we couldn't reach rawls (within timeout period). Respond with a Service Unavailable error.
+              // we couldn't reach Sam (within timeout period). Respond with a Service Unavailable error.
               case Failure(error) =>
-                respondWithErrorReport(ServiceUnavailable, ServiceUnavailable.defaultMessage, requestContext)
+                respondWithErrorReport(ServiceUnavailable, "Identity service did not produce a timely response, please try again later.", requestContext)
```

Comment:
Also here for my own clarity - it's not necessary to be consistent with an error code here due to the fact that the messages are different and can be clearly followed from log reporting.

Reply:
Yeah, I only put an error code in the two cases that shared the same message. Could standardize across all, but wasn’t sure it added value